traffic_ops/traffic_ops_golang/riak.go

```diff
@@ -0,0 +1,284 @@
+			log.Errorf("%v %v\n", r.RemoteAddr, err)
+			w.WriteHeader(status)
+			fmt.Fprintf(w, http.StatusText(status))
+		}
+		defer r.Body.Close()
+
+		ctx := r.Context()
+		pathParams, err := getPathParams(ctx)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		xmlId := pathParams["xml-id"]
+		data, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		// validate that the received data is a valid jwk keyset
+		var keySet map[string]URISignerKeyset
+		if err := json.Unmarshal(data, &keySet); err != nil {
+			log.Errorf("%v\n", err)
+			handleErr(err, http.StatusBadRequest)
+			return
+		}
+		if err := validateURIKeyset(keySet); err != nil {
+			log.Errorf("%v\n", err)
+			handleErr(err, http.StatusBadRequest)
+			return
+		}
+
+		// create a storage object and store the data
+		obj := &riak.Object{
+			ContentType:     "text/json",
+			Charset:         "utf-8",
+			ContentEncoding: "utf-8",
+			Key:             xmlId,
+			Value:           []byte(data),
+		}
+
+		err = saveObject(obj, CdnUriKeysBucket, db, cfg)
+		if err != nil {
+			log.Errorf("%v\n", err)
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", data)
+	}
+}
+
+// saves an object to riak storage
+func saveObject(obj *riak.Object, bucket string, db *sqlx.DB, cfg Config) error {
+	// create and start a cluster
+	cluster, err := getRiakCluster(db, cfg, 12)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := cluster.Stop(); err != nil {
+			log.Errorf("%v\n", err)
+		}
+	}()
+
+	if err = cluster.Start(); err != nil {
+		return err
+	}
+
+	// build store command and execute.
+	cmd, err := riak.NewStoreValueCommandBuilder().
+		WithBucket(bucket).
+		WithContent(obj).
+		Build()
+	if err != nil {
+		return err
+	}
+	if err := cluster.Execute(cmd); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// fetch an object from riak storage
+func fetchObject(key string, bucket string, db *sqlx.DB, cfg Config) (*riak.FetchValueCommand, error) {
+	// build the fetch command
+	cmd, err := riak.NewFetchValueCommandBuilder().
+		WithBucket(bucket).
+		WithKey(key).
+		Build()
+	if err != nil {
+		return nil, err
+	}
+	// create and start a riak cluster
+	cluster, err := getRiakCluster(db, cfg, 12)
+	if err != nil {
+		log.Errorf("%v\n", err)
+		return nil, err
+	}
+	defer func() {
+		if err := cluster.Stop(); err != nil {
+			log.Errorf("%v\n", err)
+		}
+	}()
+	if err = cluster.Start(); err != nil {
+		return nil, err
+	}
+	if err = cluster.Execute(cmd); err != nil {
+		return nil, err
+	}
+	fvc := cmd.(*riak.FetchValueCommand)
+
+	return fvc, err
+}
+
+// endpoint handler for fetching uri signing keys from riak
+func getUrisignkeysHandler(db *sqlx.DB, cfg Config) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		handleErr := func(err error, status int) {
```

Comment:
I think both of the handleErr functions should use handleErr := tc.GetHandleErrorFunc(w, r), which applies the alerts pattern to the error passed in.